addons/io_scene_gltf2/io/exp/gltf2_io_user_extensions.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2019 The glTF-Blender-IO authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import glob
+import imp
+import importlib
+import os.path
+import bpy
+from io_scene_gltf2.io.com.gltf2_io_extensions import Extension
+
+class GLTF2ExtensionMesh:
+    def __init__(self, blender_mesh, vertex_groups, modifiers):
+        self.blender_mesh = blender_mesh,
+        self.vertex_groups = vertex_groups,
+        self.modifiers = modifiers
+
+class GLTF2ExtensionNode:
+    def __init__(self, blender_object):
+        self.blender_object = blender_object 
+
+class GLTF2ExtensionAnimationChannelTarget:
+    def __init__(self, channels, blender_object, bake_bone):
+        self.channels = channels,
+        self.blender_object = blender_object,
+        self.bake_bone = bake_bone
+
+class GLTF2ExtensionAnimationChannels:
+    def __init__(self, channels, blender_object, bake_bone):
+        self.channels = channels,
+        self.blender_object = blender_object,
+        self.bake_bone = bake_bone
+
+class GLTF2ExtensionAnimationSamplers:
+    def __init__(self, channels, blender_object_if_armature, bake_bone, bake_channel):
+        self.channels = channels,
+        self.blender_object_if_armature = blender_object_if_armature,
+        self.bake_bone = bake_bone
+        self.bake_channel = bake_channel
+
+class GLTF2ExtensionAnimations:
+    def __init__(self, blender_action, blender_object):
+        self.blender_action = blender_action,
+        self.blender_object = blender_object
+
+class GLTF2ExtensionCameras:
```

Comment:
Grammatical number seems to be inconsistent here. Some are singular (`GLTF2ExtensionMesh`, `GLTF2ExtensionNode`, etc.) and some are plural (`GLTF2ExtensionCameras`, `GLTF2ExtensionMaterials`, etc.)

Reply:
I using the same notation as the gltf2_blender_gather_<stuff>.py files. There are some that are `cameras` and others that are `mesh`.

Reply:
Node is the only difference then (`GLTF2ExtensionNode` but `gltf2_blender_gather_nodes.py`). I still think they should still have a consistent number though.